src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -424,6 +429,15 @@ private string GetXrefHref(FilePath file, string uid, int uidCount, bool isRootL
 
                         errors.AddIfNotNull(xrefError);
 
+                        if (xrefSpec != null &&
+                        schema.XrefType != null &&
+                        schema.ContentType == JsonSchemaContentType.Xref &&
+                        (xrefSpec.SchemaType is null ||
+                        !xrefSpec.SchemaType.Equals(schema.XrefType)))
```

Comment:
Some checks are not necessary, and also use case insensitive comparision.
```suggestion
                        if (xrefSpec != null && schema.XrefType != null && !schema.XrefType.Equals(xrefSpec.SchemaType, StringComparison.OrdinalIgnoreCase)))
```

Reply:
ok.